src/core/Global.ttslua

```diff
@@ -140,6 +142,26 @@ function isDeck(x) return x.tag == 'Deck' end
 
 function isCardOrDeck(x) return x.tag == 'Card' or x.tag == 'Deck' end
 
+-- Event hook for any object search.  When chaos tokens are manipulated while the chaos bag
+-- container is being searched, a TTS bug can cause tokens to duplicate or vanish.  We lock the
+-- chaos bag during search operations to avoid this.
+function onObjectSearchStart(object, playerColor)
+  findChaosBag()
+  if object.getGUID() == chaosbag.getGUID() then
```

Comment:
Is that a feature of sced, tts, or lua?

Reply:
Little of all three.

SCED we don't do object comparisons much because object references are unreliable.  In this case we're finding the objects live (or being passed in the event) so it works.
TTS's objects are just Lua tables/objects
Lua's == is a straight value comparison.